### ISSUE TITLE
Placeholder text doesn't respect CSS styling of typed text

### ISSUE THREAD

**Issue by shanereichart** [tinymce/tinymce#6348]
The new v5.2 placeholder doesn't work well with CSS which may be applied to the real content text. In my case, the padding applied to the editor and a `::before` pseudo-element on the `<p>` element within the editor cause the real text to be positioned entirely differently than the placeholder text. (It also doesn't clear nicely on focus.)

Placeholder text:

![image](https://user-images.githubusercontent.com/16807422/102263136-f0982a80-3ee1-11eb-95c3-cc3979273c7e.png)

Real text:

![image](https://user-images.githubusercontent.com/16807422/102263528-6e5c3600-3ee2-11eb-8539-e8df568e5b5a.png)

I've initialized the TinyMCE editor on a blockquote, and here is the CSS applied to that blockquote:

```
blockquote {
	background-color: #f9f9f9;
	border: thin solid #ececec;
	padding: 20px;
	font-family: 'Old Standard TT', serif;
	font-style: italic;
	color: #181818;
	font-size: 22px;
}

blockquote p:first-of-type::before {
	content: url('lib/quote.png');
	display: inline-block;
	position: relative;
	bottom: -5px;
	margin-top: -5px;
}
```

Instead, we've resorted to a JS solution:

```
<blockquote data-placeholder="The quote content goes here."></blockquote>
...
<script>
tinymce.init({
    selector: 'blockquote',
    setup: function(editor) {
        var placeholder = $('#' + editor.id).attr('data-placeholder');
        if (typeof placeholder !== 'undefined' && placeholder !== false) {
            var is_default = false;
            editor.on('init', function() {
                var content = editor.getContent();
                if (content.length === 0) {
                    editor.setContent(placeholder);
                    content = placeholder;
                }
                is_default = (content == placeholder);
            })
            .on('focus', function() {
                var content = editor.getContent();
                if (content == placeholder) {
                    editor.setContent('');
                }
            })
            .on('blur', function() {
                if (editor.getContent().length === 0) {
                    editor.setContent(placeholder);
                }
            });
        }
    }
...
</script>
```

**Comment by lnewson** [comment]
Hi @shanereichart, the placeholder is just positioned using CSS based on the default styles, so if you add a before psudeo selector it's not going to work as you noted. You can change where it's positioned by including a rule for `.mce-content-body[data-mce-placeholder]::before`. Here's a fiddle showing it working for your use case: https://fiddle.tiny.cloud/w1haab. Doesn't that solve your issue?

**Comment by shanereichart** [comment]
Hi, @lnewson! As I investigated it, I did see that it was simple CSS. The issue for us is that we use this editor across multiple sites, and within those sites, in multiple places, each of which has it's own styling, and even that changes frequently. So ultimately, we need a dynamic solution. Since we also like having the placeholder clear on focus, our JS solution solves both issues, but it would be nice if some method of making the placeholder behave this way baked in. I suppose this is a feature request. Thanks!
